src/metabase/types.clj

```diff
@@ -5,6 +5,7 @@
 (derive :type/Dictionary :type/Collection)
 (derive :type/Array :type/Collection)
 
+(derive :type/Nil :type/*)
```

Comment:
Fair point. But having to essentially dispatch everything twice (wherever I'm using `:type/*`) seems tedious and error prone. I think it's then better to change dispatch logic to use a cond rather than trying to reuse existing `:type` ontology.